packages/ui-library/src/elements/input/index.js

```diff
@@ -0,0 +1,67 @@
+import classNames from "classnames";
+import { values } from "lodash";
+import PropTypes from "prop-types";
+
+export const INPUT_TYPES = {
+	BUTTON: "button",
+	CHECKBOX: "checkbox",
+	COLOR: "color",
+	DATE: "date",
+	DATETIME_LOCAL: "datetime-local",
+	EMAIL: "email",
+	FILE: "file",
+	HIDDEN: "hidden",
+	IMAGE: "image",
+	MONTH: "month",
+	NUMBER: "number",
+	PASSWORD: "password",
+	RADIO: "radio",
+	RANGE: "range",
+	RESET: "reset",
+	SEARCH: "search",
+	SUBMIT: "submit",
+	TEL: "tel",
+	TEXT: "text",
+	TIME: "time",
+	URL: "url",
+	WEEK: "week",
+};
+
+/**
+ * @param {string} [type="text"] The type of input. See INPUT_TYPES.
+ * @param {string} [className=""] CSS class.
+ * @param {boolean} [disabled=false] Whether the input is disabled.
+ * @param {object} [props] Optional extra properties.
+ * @returns {JSX.Element} Input component.
+ */
+const Input = ( {
+	type,
+	className,
+	disabled,
+	...props
+} ) => (
+	<input
+		type={ type }
+		className={ classNames(
+			"yst-input",
+			disabled && "yst-input--disabled",
+			className,
+		) }
```

Comment:
Manage error state here using `invalid` html attr. Adds the need for a wrapping element to position the icon.

```css
.yst-input {
  // wrapper styles
}

.yst-input__input {
  // styles now on base class?
}
```